src/tokenizer.ts

```diff
@@ -41,14 +49,19 @@ export function scanToken (state: State): Token {
     return scanCommentBlock(state);
   }
 
-  if (isString(ch)) {
-    return scanString(state);
+  if (isString(ch, dialect) && ch !== null) {
```

Comment:
Do any of the `is*` functions return true if `ch === null`? Could we just wrap all of these `if (is*(ch)) { return scan*(state); }` block with a singular `if (ch !== null) { }`?

Reply:
Yes possibly? I didn't want to make such a large change, but certainly possible